sale_stock_picking_invoicing/tests/test_sale_stock.py

```diff
@@ -0,0 +1,429 @@
+            "partner_invoice_id": self.partner.id,
+            "partner_shipping_id": self.partner.id,
+            "order_line": [
+                (
+                    0,
+                    0,
+                    {
+                        "name": self.product.name,
+                        "product_id": self.product.id,
+                        "product_uom_qty": 3.0,
+                        "product_uom": self.product.uom_id.id,
+                        "price_unit": self.product.list_price,
+                    },
+                )
+            ],
+            "pricelist_id": self.env.ref("product.list0").id,
+        }
+        self.so = self.env["sale.order"].create(so_vals)
+
+        # confirm our standard so, check the picking
+        self.so.action_confirm()
+        self.assertTrue(
+            self.so.picking_ids,
+            'Sale Stock: no picking created for "invoice on '
+            'delivery" storable products',
+        )
+
+        # set stock.picking to be invoiced
+        self.assertTrue(
+            len(self.so.picking_ids) == 1,
+            "More than one stock " "picking for sale.order",
+        )
+        self.so.picking_ids.set_to_be_invoiced()
+
+        # validate stock.picking
+        stock_picking = self.so.picking_ids
+
+        # compare sale.order.line with stock.move
+        stock_move = stock_picking.move_lines
+        sale_order_line = self.so.order_line
+
+        sm_fields = [key for key in self.env["stock.move"]._fields.keys()]
+        sol_fields = [key for key in self.env["sale.order.line"]._fields.keys()]
+
+        skipped_fields = [
+            "id",
+            # 'S00029/FURN_7777: Stock>Customers' != 'S00029 - Office Chair'
+            "display_name",
+            "state",
+            # Price Unit in move is different from sale line
+            # TODO: Should be equal? After Confirmed stock picking
+            #  the value will be change based Stock Valuation
+            #  configuration.
+            "price_unit",
+            # There are a diference for field Name
+            # '[FURN_7777] Office Chair' != 'Office Chair'
+            "name",
+        ]
+        common_fields = list(set(sm_fields) & set(sol_fields) - set(skipped_fields))
+
+        for field in common_fields:
+            self.assertEqual(
+                stock_move[field],
+                sale_order_line[field],
+                "Field %s failed to transfer from "
+                "sale.order.line to stock.move" % field,
+            )
+
+    def test_picking_sale_order_product_and_service(self):
+        """
+        Test Sale Order with product and service
+        """
+
+        sale_order_2 = self.env.ref(
+            "sale_stock_picking_invoicing.main_company-sale_order_2"
+        )
+        # Necessary to get the currency
+        sale_order_2.onchange_partner_id()
+        sale_order_2.action_confirm()
+        self.assertTrue(sale_order_2.state == "sale")
```

Comment:
a lot of these assertions seem to be testing Odoo native functionality and not relevant to this module, only things that are impacted by this module should be tested

Reply:
I tried to remove all redundant tests